pandas/core/frame.py

```diff
@@ -8738,6 +8741,30 @@ def blk_func(values):
 
         if constructor is not None:
             result = self._constructor_sliced(result, index=labels)
+
+        check_func = (
+            lambda x: is_extension_array_dtype(x)
+            or is_object_dtype(x)
+            or needs_i8_conversion(x)
+        )
+        msg = (
+            "In a future version, DataFrame reduction operations with "
+            "ExtensionDtype or ObjectDtype will be performed column-wise. "
+            "To keep the old behavior, explicitly cast columns to bool/numeric "
+            "dtypes."
+        )
+        if axis == 0:
+            if self.dtypes.apply(check_func).any():
```

Comment:
Do we know of other ops where this is a problem, or only `any`/`all` ? Otherwise I would also explicitly check this first.